projects/batfish-common-protocol/src/main/java/org/batfish/storage/FileStorageImpl.java

```diff
@@ -0,0 +1,159 @@
+package org.batfish.storage;
+
+import com.google.common.collect.Sets;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.batfish.common.BatfishException;
+import org.batfish.common.BfConsts;
+import org.batfish.common.util.CommonUtil;
+import org.batfish.datamodel.pojo.Analysis;
+
+
+public class FileStorageImpl implements Storage {
+
+  private final Path _containersLocation;
+
+  public FileStorageImpl(Path containersLocation) throws BatfishException {
+    try {
+      if (containersLocation != null) {
+        _containersLocation = containersLocation;
+        _containersLocation.toFile().mkdir();
+      } else {
+        throw new BatfishException("container location is null");
+      }
+    } catch (InvalidPathException e) {
+      throw new BatfishException("cannot resolve containers location '" + containersLocation + "'");
+    }
+  }
+
+  /**
+   * Retrieve a container Path
+   *
+   * @param containerName Container name
+   * @return Path of the container
+   */
+  // deprecated
+  @Override
+  public Path getContainerPath(String containerName) {
+    Path containerDir = _containersLocation.resolve(containerName).toAbsolutePath();
+    if (!Files.exists(containerDir)) {
+      throw new BatfishException("Container '" + containerName + "' not found");
+    }
+    return containerDir;
+  }
+
+  @Override
+  public Analysis getAnalysis(String containerName, String analysisName) {
+    Path aDir =
+        _containersLocation.resolve(
+            Paths.get(containerName, BfConsts.RELPATH_ANALYSES_DIR, analysisName));
+    if (!Files.exists(aDir)) {
+      throw new BatfishException(
+          String.format(
+              "Analysis '%s' doesn't exist for container '%s'", analysisName, containerName));
+    }
+    Path questionsDir = aDir.resolve(BfConsts.RELPATH_QUESTIONS_DIR);
+    if (!Files.exists(questionsDir)) {
+      throw new BatfishException(
+          String.format("Analysis '%s' doesn't contain questions directory", analysisName));
+    }
+    Map<String, String> questions = new HashMap<>();
+    for (Path questionDir : CommonUtil.getEntries(questionsDir)) {
+      Path questionFile = questionDir.resolve(BfConsts.RELPATH_QUESTION_FILE);
+      if (!Files.exists(questionFile)) {
+        throw new BatfishException(
+            String.format(
+                "Failed to read analysis '%s', question empty: '%s'",
+                analysisName, questionDir.getFileName()));
+      }
+      questions.put(
+          questionDir.getFileName().toString(),
+          CommonUtil.readFile(questionDir.resolve(BfConsts.RELPATH_QUESTION_FILE)));
```

Comment:
Use 'questionFile' instead of 'questionDir.resolve(BfConsts.RELPATH_QUESTION_FILE)'?